HelloTorchIOS/HelloTorchIOS/StrobeLights.swift

```diff
@@ -1,9 +1,39 @@
-//
-//  StrobeLights.swift
-//  HelloTorchIOS
-//
-//  Created by Luis Valle on 02/05/2018.
-//  Copyright © 2018 Novoda. All rights reserved.
-//
-
 import Foundation
+import AVFoundation
+
+class StrobeLights {
+    
+    let lightsController = LightsLogic()
+    var counter: Int = 0
+    var timer: Timer = Timer()
+    var frequency: Double = 500
+    var start = DispatchTime.now()
+    var end = DispatchTime.now()
+    var isCounting = false
+    
+    // Start Strobe process
+    func toggleStrobe () {
+        if (self.isCounting == true) {
+            print("Turning timer off")
+            self.isCounting = false
+            self.timer.invalidate()
+            
+            self.end = DispatchTime.now()
+            let nanoTime = end.uptimeNanoseconds - start.uptimeNanoseconds
+            let timeInterval = Double(nanoTime) / 1_000_000_000
+            print("I counted this high \(counter) in this many seconds \(timeInterval) ")
+            counter = 0
```

Comment:
I think this might be better abstracted into a private method in the same class, so you can read: 
```
if isCounting {
   turnTimerOff() 
   endDispatchTime() 
   resetCounter() 
}
``` 
And a similar abstraction for the else statement. How this strobe light works is a bit unclear since we have nit used it before so I think proper methods naming could help someone else pick up this code later on.

Reply:
Also the part of the code of: 

```
let nanoTime = end.uptimeNanoseconds - start.uptimeNanoseconds
let timeInterval = Double(nanoTime) / 1_000_000_000
```

Could be set as a `print("I counted this high \(counter) in this many seconds \(timeInterval()) ")`. Working with time and date is tricky and separating it would clean up the code since it has nothing to do with the light strobe.

Reply:
I can tell you're loving the Clean Code book :)

Reply:
I am loving it quite a lot 😄